### DISCUSSION TITLE
Anyone got the custom config working with @vscode/test-electron?

### DISCUSSION THREAD

**Discussion by PEZ** [microsoft/vscode-discussions#392] [Extension Development QnA] [answered]
I fail to set up a custom vscode insiders instance where I can install an extension for my tests. Pasting my issue description from:

* https://github.com/microsoft/vscode-test/issues/183

> Following the instructions for Custom setup of VS Code extension integration tests here: https://code.visualstudio.com/api/working-with-extensions/testing-extension#custom-setup-with-vscodetestelectron
> 
> It does not work as advertised. At least not when using it to download and run latest Insiders. There are several problems:
> 
> `resolveCliPathFromVSCodeExecutablePath` resolves to the wrong path.
> 
> * It resolves to `<workspace-root>/.vscode-test/vscode-darwin-arm64-insiders/Visual Studio Code - Insiders.app/Contents/MacOS/bin/code`
> * There is no such file. There is, however a `code` executable in `<wokspace-root>/.vscode-test/vscode-darwin-arm64-insiders/Visual Studio Code - Insiders.app/Contents/Resources/app/bin/code`
> * The path returned when downloading the executable reports is: `<workspace-root>/.vscode-test/vscode-darwin-arm64-insiders/Visual Studio Code - Insiders.app/Contents/MacOS/Electron`
> 
> This causes the call to `cp.spawnSync` to fail silently. Giving it the path to the `code` executable in `Resources` ”works” as far as I can tell. I say ”works”, because even though it is spawned, it is not used for running tests. Instead `runTests()` spawns a new VS Code instance, with some other executable (I don't know which one).
> 
> * `@vscode/test-electron` Version: 2.2.1

Anyone else who has ran into problems with this and have found a workaround? I actually only need to install an extension, so if there's a way to do that without the custom setup, that would work too for me.

**Comment by isidorn** [comment]
@connor4312 might be able to help here

**Reply by PEZ** [reply]
Thanks, @isidorn! And tanks @connor4312 for addressing [the issue](https://github.com/microsoft/vscode-test/issues/187) I posted with a fix. Though either the issue was only partly fixed, or I am doing something wrong, because it's still not quite working:

* https://github.com/microsoft/vscode-test/issues/187

I _think_ I am pretty much following [the recipe](https://code.visualstudio.com/api/working-with-extensions/testing-extension#custom-setup-with-vscodetestelectron) as outlined.

**Comment by PEZ** [comment, marked as answer]
@connor4312 in https://github.com/microsoft/vscode-test/issues/187#issuecomment-1409078049:

> By default, tests run with a different user data directory and extensions directory than the global system one. This can be disabled by setting `reuseMachineInstall: true` in the options to runTests. Alternatively, use `resolveCliArgsFromVSCodeExecutablePath` instead of `resolveCliPathFromVSCodeExecutablePath` and include those arguments in your cli call to set the user data dir / extensions dir correctly. Ctrl+f for "python" in the [readme](https://github.com/microsoft/vscode-test/blob/main/README.md) for an example of doing exactly this.

For me `reuseMachineInstall: true` worked perfectly. Thanks!
